src/legacy/index.ts

```diff
@@ -123,6 +125,7 @@ async function buildGraph(
     const depPkg: types.PkgInfo = {
       name: depName,
       version: dep.version,
+      purl: dep.purl,
     };
 
```

Comment:
If we want to avoid churn on the data, maybe something like:
```suggestion
    };
    if (dep.purl) {
      depPkg.purl = dep.purl;
    }
```

Reply:
Isn't that the exact same thing?
`depPkg.purl` is an optional field on `PkgInfo`, so it will default to `undefined`. Whether we explicitly set it to `undefined` by doing `dep.purl` shouldn't matter? 🤔

Reply:
It's not the same thing at runtime. In one case the key exists (and has the value `undefined`) and in the other the key doesn't exist at all

Reply:
Ah yes, because that's not the only place where I need to adjust that.

In that case, all good. Also no changes to the snapshots anymore.